warehouse/oidc/forms/gitlab.py

```diff
@@ -0,0 +1,111 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+import wtforms
+
+from warehouse import forms
+from warehouse.i18n import localize as _
+from warehouse.oidc.forms._core import PendingPublisherMixin
+
+# TODO Gitlab also accepts emojis
+# see:
+# https://docs.gitlab.com/ee/user/reserved_names.html#limitations-on-project-and-group-names
```

Comment:
Oof. I think we can either say "emoji are not supported, regardless of what GitLab lets you do" or use a property escape like `\p{Emoji_Presentation}`, although I don't know if Python's `re` supports that.

But probably just the first idea 🙂

Reply:
Yeah I'd also go with not supporting emojis for now. We can always add support for them later if needed

Reply:
I'm a bit confused by this, because glancing at https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/path_regex.rb (referenced from this link) it doesn't seem like Gitlab permits emoji either.

I think it's fine for us to not support emoji for now.